File: 0704-binary-search/0704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    let left = 0;
    let right = nums.length-1;

    while (left <= right) {
       let mid = Math.round((left + right) / 2);
       if (nums[mid] === target) {
            return mid;
       } else if (nums[mid] > target) {
           right = mid - 1;
       } else {
            left = mid + 1;
       }
    }
    
    return -1;
};

File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0, right = nums.length-1;

        while (left + 1 < right) {
            int mid = left + (right - left)/2;
            if (nums[mid] < target) {
                left = mid;
            } else {
                right = mid;
            }
        }
        // start, end

        if (nums[right] == target) return right;
        if (nums[left] == target) return left;
        return -1;
    }
}